util/check-build.sh

```diff
@@ -0,0 +1,149 @@
+else
+    prerequisites=(git curl make cmake npm go gvm golangci-lint wasm2wat clang gotestsum yarn)
+fi
+
+for pkg in "${prerequisites[@]}"; do
+    if command_exists "$pkg"; then
+        # There is no way to check for wabt / llvm directly, since they install multiple tools
+        # So instead, we check for wasm2wat and clang, which are part of wabt and llvm respectively
+        # and if they are installed, we assume wabt / llvm is installed else we ask the user to install wabt / llvm
+        [[ "$pkg" == "wasm2wat" ]] && pkg="wabt"
+        [[ "$pkg" == "clang" ]] && pkg="llvm"
+
+        # Check for specific symbolic links related to wasm-ld on Linux and macOS
+        if [[ "$pkg" == "llvm" ]]; then
+            if [[ "$OS" == "Linux" ]]; then
+                if [ ! -L /usr/local/bin/wasm-ld ]; then
+                    echo -e "${YELLOW}Creating symbolic link for wasm-ld on Linux.${NC}"
+                    sudo ln -s /usr/bin/wasm-ld-13 /usr/local/bin/wasm-ld
+                else
+                    echo -e "${GREEN}Symbolic link for wasm-ld on Linux is already present.${NC}"
+                fi
+            elif [[ "$OS" == "Darwin" ]]; then
+                if [ ! -L /usr/local/bin/wasm-ld ]; then
+                    echo -e "${YELLOW}Creating symbolic link for wasm-ld on macOS.${NC}"
+                    sudo mkdir -p /usr/local/bin
+                    sudo ln -s /opt/homebrew/opt/llvm/bin/wasm-ld /usr/local/bin/wasm-ld
+                else
+                    echo -e "${GREEN}Symbolic link for wasm-ld on macOS is already present.${NC}"
+                fi
+            fi
+        fi
+
+        echo -e "${GREEN}$pkg is installed.${NC}"
+    else
+        [[ "$pkg" == "wasm2wat" ]] && pkg="wabt"
+        [[ "$pkg" == "clang" ]] && pkg="llvm"
+        echo -e "${RED}$pkg is not installed. Please install $pkg. $INSTALLATION_DOCS_URL${NC}"
+        exit 1
+    fi
+done
+
+# Step 6: Check Node.js version
+if command_exists node && node -v | grep -q "v18"; then
+    echo -e "${GREEN}Node.js version 18 is installed.${NC}"
+else
+    echo -e "${RED}Node.js version 18 not installed. $INSTALLATION_DOCS_URL${NC}"
+    exit 1
+fi
+
+# Step 7a: Check Rust version
+if command_exists rustc && rustc --version | grep -q "1.80.1"; then
+    echo -e "${GREEN}Rust version 1.80.1 is installed.${NC}"
+else
+    echo -e "${RED}Rust version 1.80.1 not installed. $INSTALLATION_DOCS_URL${NC}"
+    exit 1
+fi
+
+# Step 7b: Check Rust nightly toolchain
+if rustup toolchain list | grep -q "nightly"; then
+    echo -e "${GREEN}Rust nightly toolchain is installed.${NC}"
+else
+    echo -e "${RED}Rust nightly toolchain is not installed. Install it using: rustup toolchain install nightly${NC}"
+    exit 1
+fi
+
+# Step 8: Check Go version
+if command_exists go && go version | grep -q "go1.23"; then
+    echo -e "${GREEN}Go version 1.23 is installed.${NC}"
```

Comment:
https://docs.arbitrum.io/run-arbitrum-node/nitro/build-nitro-locally actually calls for go1.21

I wonder if we should read the version from the `go.mod` file and just make sure that the actual version of go is in the same major version? That way, we wouldn't have to update the script when we update `go.mod`?

Reply:
Done.